File: android/app/src/main/kotlin/com/example/native_channel/plugin/DeviceInfoApiImpl.kt
package com.example.native_channel.plugin

import DeviceInfo
import DeviceInfoAPI

import android.os.Build


class DeviceInfoApiImpl : DeviceInfoAPI {
    override fun getDeviceInfo(): DeviceInfo {

        val deviceName: String = Build.MODEL
        val deviceManufacturer: String = Build.MANUFACTURER
        val deviceVersion: String = Build.VERSION.RELEASE
        val deviceFamily: String = Build.DEVICE

        return DeviceInfo(
            deviceFamily = deviceFamily,
            deviceName = deviceName,
            deviceVersion = deviceVersion,
            deviceModel = deviceManufacturer,
        )
    }

}

File: ios/Runner/Plugin/DeviceInfoApiImpl.swift
//
//  DeviceInfoApiImpl.swift
//  Runner
//
//  Created by Ashwin Shrestha on 24/01/2025.
//
import UIKit

class DeviceInfoApiImpl : DeviceInfoAPI {
    
     func getDeviceInfo() -> DeviceInfo {
        let device = UIDevice.current
        let deviceInfo: DeviceInfo = DeviceInfo(
            deviceName: device.name,
            deviceModel: device.model,
            deviceVersion: device.systemVersion,
            deviceFamily: "iOS"
        )
         print("Device Info: \(deviceInfo)")
        return deviceInfo
      }
}
